### PR TITLE
Feature/plcn 159 add startup project

### PR DESCRIPTION
**Pull request by Nicolai-Bak**

### CODE REVIEW

File: Pelican.App/Program.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+builder.Services.AddApi();
+
+//Adding GraphQl Specific extensions and services.
+builder.Services
+	.AddPresentationGraphQL()
+	.AddDataLoaders();
+
+var app = builder.Build();
+
+app.UseApi();
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.App/Program.cs:21]: The Pelican.App project is added In an attempt to follow the principles of Clean Architecture. 
By adding the new project as the startup project it is possible to remove infrastructure dependencies from the presentation layer as it was only used to add services and middleware.
In addition, it removes the doubt as to which Presentation project the program file belongs

File: Pelican.Domain/Settings/HubSpotAppSettings.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿namespace Pelican.Infrastructure.HubSpot.Settings;
+﻿namespace Pelican.Domain.Settings;
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Domain/Settings/HubSpotAppSettings.cs:1]: Both presentation.api and infrastructure.hubspot needs access to the settings and as they do not depend on each other anymore the settings was moved

**Review by AntonEsbjergIT-Minds — CHANGES_REQUESTED** [review]
I have created the first two changes myself but. the last one about renaming you could do if you think it should be

File: Pelican.Architecture.Test/ProjectReferencesTests.cs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+	private const string APP_NAMESPACE = "Pelican.App";
+	private const string DOMAIN_NAMESPACE = "Pelican.Domain";
+	private const string APPLICATION_NAMESPACE = "Pelican.Application";
+	private const string INFRASTRUCTURE_HUBSPOT_NAMESPACE = "Pelican.Infrastructure.HubSpot";
+	private const string INFRASTRUCTURE_PERSISTENCE_NAMESPACE = "Pelican.Infrastructure.Persistence";
+	private const string PRESENTATIION_API_NAMESPACE = "Pelican.Presentation.Api";
+	private const string PRESENTATIION_GRAPHQL_NAMESPACE = "Pelican.Presentation.GraphQL";
+
+	[Fact]
+	public void Domain_ShouldNotDependOn()
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Architecture.Test/ProjectReferencesTests.cs:16]: "_IsSuccessful" added

File: Pelican.Architecture.Test/ProjectReferencesTests.cs [hunk 4]
```diff
@@ -0,0 +1,169 @@
+﻿using NetArchTest.Rules;
+using Xunit;
+
+namespace Pelican.Architecture.Test;
+
+public class ProjectReferencesTests
+{
+	private const string APP_NAMESPACE = "Pelican.App";
+	private const string DOMAIN_NAMESPACE = "Pelican.Domain";
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Architecture.Test/ProjectReferencesTests.cs]: removed

File: Pelican.App/Program.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (llh-itm) [reply Pelican.App/Program.cs:23]: that makes sense, in that case this seems like a good solution then:)

File: Pelican.Application/Pelican.Application.csproj [hunk 5]
```diff
@@ -11,7 +11,6 @@
 		<PackageReference Include="GreenDonut" Version="12.15.1" />
 		<PackageReference Include="HotChocolate.Types" Version="12.15.1" />
 		<PackageReference Include="MediatR" Version="11.0.0" />
```
Comment (llh-itm) [inline Pelican.Application/Pelican.Application.csproj:13]: is mediatR used in this project?

File: Pelican.Infrastructure.HubSpot/Pelican.Infrastructure.HubSpot.csproj [hunk 6]
```diff
@@ -7,7 +7,6 @@
 	</PropertyGroup>
 
 	<ItemGroup>
-		<PackageReference Include="Microsoft.Extensions.Hosting" Version="6.0.1" />
 		<PackageReference Include="Microsoft.Extensions.Http" Version="6.0.0" />
```
Comment (llh-itm) [inline Pelican.Infrastructure.HubSpot/Pelican.Infrastructure.HubSpot.csproj:10]: looks like Microsoft.Extensions.Http can be removed as well

File: Pelican.Domain/Settings/HubSpotAppSettings.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (llh-itm) [reply Pelican.Domain/Settings/HubSpotAppSettings.cs:1]: that makes sense, in that case this seems like a good solution then:)

File: Pelican.App/Program.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (llh-itm) [reply Pelican.App/Program.cs:23]: that makes sense, in that case this seems like a good solution then:)

File: Pelican.Application/Pelican.Application.csproj (hunk 5 above)
Comment (Nicolai-Bak) [reply Pelican.Application/Pelican.Application.csproj:13]: MediatR is used to implement Query- and CommandHandlers

File: Pelican.Infrastructure.HubSpot/Pelican.Infrastructure.HubSpot.csproj (hunk 6 above)
Comment (Nicolai-Bak) [reply Pelican.Infrastructure.HubSpot/Pelican.Infrastructure.HubSpot.csproj:10]: true, removed now

File: Pelican.Application/Pelican.Application.csproj (hunk 5 above)
Comment (llh-itm) [reply Pelican.Application/Pelican.Application.csproj:13]: ahh i see

**Review by llh-itm — APPROVED** [review]

**Pull request merged**